### PR TITLE
feat: unify and simplify welcome and landing

### LINKED ISSUE #36479: New welcome page
Mockup and proposal:

- [ ] Create and share the copy of the new proposed welcome page. ~ @QuincyLarson 

Implementation:

- [ ] Todo: Add stories and implement... @ahmadabdolsaheb

### LINKED ISSUE #36480: New landing page
Mockup and proposal:

- [ ] Create and share the copy of the new proposed landing page. ~ @QuincyLarson 

Implementation:

- [ ] Todo: Add stories and implement... @ahmadabdolsaheb

### PR DESCRIPTION
**Pull request by ahmaxed**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

This PR attempts to unify and simplify landing and welcome page.
It will alter between the legacy welcome and landing pages based on authentication.

Closes #36480 
Closes #36479

### CODE REVIEW

File: client/src/pages/index.js [hunk 1]
```diff
@@ -1,243 +1,111 @@
...
+            </Col>
+          </Row>
+          <Spacer />
+          <BigCallToAction />
+          <Spacer />
+          <Spacer />
           <Image
-            alt="Maxim Orlov's testimonial image"
-            className='testimonial-image img-center'
+            alt='companies featuring freeCodeCamp'
```
Comment (raisedadead) [inline client/src/pages/index.js]:
```suggestion
            alt='companies featuring freeCodeCamp.org'
```

File: client/src/pages/index.js [hunk 2]
```diff
@@ -1,243 +1,111 @@
...
-      <Spacer />
-      <Spacer />
-    </Grid>
-  </Fragment>
-);
+          <Spacer />
+          <Row>
+            <Col sm={10} smOffset={1} xs={12}>
+              <h2 className=' medium-heading'>
+                Since 2014, more than 40,000 freeCodeCamp graduates have gotten
```
Comment (raisedadead) [inline client/src/pages/index.js]:
```suggestion
                Since 2014, more than 40,000 freeCodeCamp.org graduates have gotten
```

**Review by raisedadead — CHANGES_REQUESTED** [review]
Preliminary branding comments. @QuincyLarson please confirm. I remember you mentioned, we should brand as freeCodeCamp.org

**Review by raisedadead — CHANGES_REQUESTED** [review]
@raisedadead it is a client only route. 
Gatsby will make any react component /pages a page. So we should make it a directory or move it to client-only-routes.

**Comment by ValeraS** [comment]
IMHO? `Welcome` component could be without any props. Checks for `acceptedPrivacyTerms` we can place into `IndexPage`. All other checks already in it.

And we can place this component into `src/components`.

**Comment by raisedadead** [comment]
@ahmadabdolsaheb I think you meant to quote my reply and accidentally edited it.

@ValeraS has the proposed the correct approach in their comment above.

**Review by raisedadead — CHANGES_REQUESTED** [review]
- [x] Implement as per https://github.com/freeCodeCamp/freeCodeCamp/pull/36669#issuecomment-524828840
- [x] Add tests for both logged in and logged out users.

File: client/src/components/layouts/Default.js [hunk 3]
```diff
@@ -157,11 +154,8 @@ class DefaultLayout extends Component {
           <style>{fontawesome.dom.css()}</style>
         </Helmet>
         <WithInstantSearch>
-          <Header disableSettings={landingPage} />
-          <div
-            className={`default-layout
-          ${landingPage ? 'landing-page' : ''}`}
-          >
+          <Header disableSettings={!isSignedIn} />
```
Comment (ValeraS) [inline client/src/components/layouts/Default.js]: It also removes the search bar from the header for not logged in users.

File: client/utils/gatsby/layoutSelector.js [hunk 4]
```diff
@@ -11,14 +11,7 @@ export default function layoutSelector({ element, props }) {
...
   } = props;
-  if (pathname === '/') {
-    return (
-      <DefaultLayout landingPage={true} pathname={pathname}>
-        {element}
-      </DefaultLayout>
-    );
-  }
-  if (element.type === FourOhFourPage) {
+  if (pathname === '/' || element.type === FourOhFourPage) {
```
Comment (ValeraS) [inline client/utils/gatsby/layoutSelector.js]:
```suggestion
  if (element.type === FourOhFourPage) {
```

File: client/src/components/welcome/index.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            </Col>
+          </Row>
+          <Spacer />
+          <Row>
+            <Col sm={10} smOffset={1} xs={12}>
+              <Button
+                block={true}
+                bsSize='lg'
+                bsStyle='primary'
+                href={`/learn`}
```
Comment (ValeraS) [inline client/src/components/welcome/index.js]: For internal links use a `Link` helper component or `navigate` function from Gatsby.

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: client/src/components/layouts/Default.js (hunk 3 above)
Comment (ahmadabdolsaheb) [reply client/src/components/layouts/Default.js]: Once the roll out the universal nav which would be similar to nav news, I believe there would not be a harm in having the search available all the time

**Comment by ahmadabdolsaheb** [comment]
@ValeraS @raisedadead this should be ready to go

**Comment by ahmadabdolsaheb** [comment]
only uploading all images to our cdn and versioning and unifying the links has remained.
currently I load the new as-seen-on.png from cdn-media-1.freecodecamp.org/learn/ which I could address in either this or with all other cdn links in another pr

File: client/src/components/layouts/Default.js (hunk 3 above)
Comment (ValeraS) [reply client/src/components/layouts/Default.js]: then while we have not done this, let's remove this check and will show search bar all the time:

https://github.com/freeCodeCamp/freeCodeCamp/blob/7065ace345f61f1ec628889ec155bc89e29f1a52/client/src/components/Header/index.js#L23

**Comment by raisedadead** [comment]
Changes requested by Valeriy.

File: client/src/components/layouts/Default.js (hunk 3 above)
Comment (ahmadabdolsaheb) [reply client/src/components/layouts/Default.js]: That for catching that @ValeriaS.
the pr has been updated accordingly.

File: client/src/components/welcome/Welcome.test.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
+/* global expect */
+import React from 'react';
+import Enzyme from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+import { IndexPage } from '../../pages';
+
+Enzyme.configure({ adapter: new Adapter() });
```
Comment (raisedadead) [inline client/src/components/welcome/Welcome.test.js]: We are moving away from `Enzyme`

File: client/src/pages/index.js [hunk 7]
```diff
@@ -1,243 +1,133 @@
...
-        </Col>
-        <Col md={4} sm={12}>
+        <Spacer />
+        <Grid className='medium-heading'>
+          <Row>
+            <Col sm={10} smOffset={1} xs={12}>
+              <h1 className='big-heading'>Welcome to freeCodeCamp</h1>
+              <Spacer />
+              <h2 className='medium-heading'>Learn to code.</h2>
+              <Spacer />
```
Comment (raisedadead) [inline client/src/pages/index.js]: Remove these `<Spacer />` between the `<h2 />` elements

File: client/src/pages/index.js [hunk 8]
```diff
@@ -1,243 +1,133 @@
...
+              </h2>
+              <Spacer />
+              <h2 className='medium-heading'>
+                It's all 100% free thanks to our nonprofit's donors.
+              </h2>
+            </Col>
+          </Row>
+          <Spacer />
+          <BigCallToAction />
+          <Spacer />
```
Comment (raisedadead) [inline client/src/pages/index.js]: Use `size` prop

**Review by raisedadead — CHANGES_REQUESTED** [review]

**Review by raisedadead — COMMENTED** [review]
Is moving away from the center aligned text for H1, H2 and other elements intentional? I thought we were not touching those. IMHO, it appears weird in certain places where we have short lines.

File: client/src/components/welcome/index.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+          </Row>
+          <Spacer />
+          <Row>
+            <Col sm={10} smOffset={1} xs={12}>
+              <Link className='btn btn-lg btn-primary btn-block' to='/learn'>
+                Build Projects and Earn Certifications
+              </Link>
+              <Link
+                className='btn btn-lg btn-primary btn-block'
+                to='settings/#portfolio-settings'
```
Comment (raisedadead) [inline client/src/components/welcome/index.js]: I don't think this anchor text works. Can you fix add some tests?

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: client/src/components/welcome/index.js [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+                className='btn btn-lg btn-primary btn-block'
+                external={true}
+                to='/news'
+              >
+                Read Developer News
+              </Link>
+              <Link
+                className='btn btn-lg btn-primary btn-block'
+                external={true}
+                to='/forum`'
```
Comment (raisedadead) [inline client/src/components/welcome/index.js]: Remove the back-tick

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: client/src/components/layouts/learn.css [hunk 11]
```diff
@@ -2,22 +2,14 @@
   height: calc(100vh - var(--header-height));
 }
 
-#learn-app-wrapper .reflex-container.vertical .reflex-splitter:hover {
```
Comment (ValeraS) [inline client/src/components/layouts/learn.css]: @ahmadabdolsaheb Could you remove this file from this PR. These changes have some issues and I fixed it in this PR #36736

Comment (ahmadabdolsaheb) [reply client/src/components/layouts/learn.css]: @ValeraS sure :)

**Comment by ahmadabdolsaheb** [comment]
@ValeraS should I also make the landing page an independent component like the welcome component?

That way index page will be the only connected component that will render different components based on state.


Hi Valery. for the indexpage, should I also isolate the landing page and make it its own component

**Comment by ValeraS** [comment]
> Hi Valery. for the indexpage, should I also isolate the landing page and make it its own component

sure, do it.

File: client/src/components/welcome/index.js (hunk 9 above)
Comment (raisedadead) [reply client/src/components/welcome/index.js]: On a second thought, lets just update this to `/settings`.

**Comment by ahmadabdolsaheb** [comment]
@raisedadead, this should be good to go.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**